qa/logging-config/src/test/java/org/elasticsearch/common/logging/ESJsonLayoutTests.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.logging;
+
+
+import org.elasticsearch.test.ESTestCase;
+import org.hamcrest.Matchers;
+import org.junit.BeforeClass;
+
+/*
+  Because this test is testing ESJsonLayout in isolation, ESMessageField pattern converter is not configured
+  Status logger is printing out warnings about this to std error (causing tests to fail)
+ */
+public class ESJsonLayoutTests extends ESTestCase {
+    @BeforeClass
+    public static void initNodeName() {
```

Comment:
I need to make sure that `NodeNamePatternConverter.setNodeName` is called only once. This approach is a bit simplistic, there are junit @rule tools that would help with too. Should this be improved?